### PR TITLE
Feature/milk inventort 2

### LINKED ISSUE #59: Issue: Introduce Milk Inventory Model Concept
***Description:***
In the ongoing development of the Django Dairy Backend, we propose the introduction of a Milk Inventory Model. This model is intended to manage and record information related to milk inventory, providing a structured approach to track milk production, storage, and usage.

---

**Milk Inventory Model Concept:**

**Fields:**
- `cow` (ForeignKey to Cow): The cow from which the milk is obtained.
- `date_of_production` (DateTime, auto_now_add=True): The date and time when the milk is produced.
- `quantity` (DecimalField): The quantity of milk produced.
- `storage_location` (CharField): The location where the milk is stored.
- `usage` (CharField): The purpose for which the milk is used (e.g., sale, processing).

**Validation:**
- Ensure that the quantity of milk is a positive number.
- Validate the presence of a storage location.
- Validate the presence of usage information.

**Discussion Points:**

**Data Integrity:**
- Emphasize the importance of validation checks for maintaining accurate milk inventory data.
- Discuss any challenges or considerations related to data validation.

**Feedback:**
- Encourage team members to provide feedback on the proposed Milk Inventory Model concept.

---

**Next Steps:**
1. Review the proposed Milk Inventory Model concept and provide feedback.
2. Discuss any additional features or modifications needed.
3. Collaborate on refining and finalizing the Milk Inventory Model.

**Note:**
This is a conceptual proposal, and feedback from the team is crucial for shaping the final implementation of the Milk Inventory Model in the Django Dairy Project.

---

Your input and suggestions for further refinement are highly appreciated.

---

@peter-evance

### PR DESCRIPTION
**Pull request by mamesareagan**
resolves #59 

This pull request introduces the following enhancements to the dairy system:

    MilkPool Model:
        Added a new model MilkPool to represent the total amount of milk in stock.
        Attributes include total_amount_in_kgs (Decimal) and last_update (DateTime).
        Implemented a __str__ method for a human-readable string representation.
    MilkRecordChangeHistory Model:
        Introduced a model MilkRecordChangeHistory to track the change history of a Milk record in terms of the total amount of milk.
        Attributes include milk (ForeignKey to Milk), previous_amount_in_kgs (Decimal), and timestamp (DateTime).
    MilkPoolUpdateHistory Model:
        Added a model MilkPoolUpdateHistory to represent the update history of the Milk Pool.
        Attributes include amount_in_kgs (Decimal) and date (Date).
   Serializers:
      Implemented serializers for the new models to facilitate data representation.

   ViewSet:

       Added a MilkPoolViewSet to handle listing operations for the MilkPool model.

### CODE REVIEW

File: inventory/urls.py [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
 app_name = "inventory"
 
 
 router = routers.DefaultRouter()
 router.register(r'cow-inventory', CowInventoryViewSet, basename='cow-inventory')
 router.register(r'cow-inventory-history', CowInventoryUpdateHistoryViewSet, basename='cow-inventory-history')
+router.register(r'milk-pool', MilkPoolViewSet, basename='milk-pool')
 
 
 urlpatterns = [
```
Comment (peter-evance) [inline inventory/urls.py:19]: More URL patterns should be included here too for `MilkPoolUpdateHistory` and `MilkRecordChangeHistory` viewsets

File: production/choices.py [hunk 2]
```diff
@@ -56,3 +56,62 @@ def lactation_stage(self, lactation):
     LATE = "Late"
     DRY = "Dry"
     ENDED = "Ended"
+
+class UsageChoices(models.TextChoices):
```
Comment (peter-evance) [inline production/choices.py:60]: Remove this, or comment it out. For now its unnecessary.

File: production/choices.py [hunk 3]
```diff
@@ -56,3 +56,62 @@ def lactation_stage(self, lactation):
...
+        ```
+    """
+
+    SALE = "Sale"
+    PROCESSING = "Processing"
+    FERMENTATION = "Fermentation"
+    OTHER = "Other"
+
+
+class StorageLocationChoices(models.TextChoices):
```
Comment (peter-evance) [inline production/choices.py:90]: Remove this, or comment it out. For now its unnecessary too.

File: production/models.py [hunk 4]
```diff
@@ -136,6 +139,8 @@ class Meta:
     cow = models.ForeignKey(Cow, on_delete=models.CASCADE, related_name="milk_records")
     amount_in_kgs = models.DecimalField(default=0, max_digits=4, decimal_places=2)
     lactation = models.ForeignKey(Lactation, on_delete=models.CASCADE, null=True, editable=False)
+    # storage_location = models.CharField(max_length=20, choices=StorageLocationChoices.choices, default=StorageLocationChoices.OTHER)
```
Comment (peter-evance) [inline production/models.py:142]: Clean up this commented out stuff so that confusion is minimised for now.

File: production/serializers.py [hunk 5]
```diff
@@ -85,6 +85,8 @@ class MilkSerializer(serializers.ModelSerializer):
     - `cow`: A nested serializer field representing the cow associated with the milk record.
     - `amount_in_kgs`: A decimal field representing the amount of milk produced in kilograms.
     - `lactation`: A nested serializer field representing the associated lactation record.
+    - `storage_location`: A choice field representing the location where the milk is stored.
+    - `usage`: A choice field representing the purpose for which the milk is used.
```
Comment (peter-evance) [inline production/serializers.py:89]: Update the documentation here to be correct

File: production/validators.py [hunk 6]
```diff
@@ -180,3 +181,21 @@ def validate_cow_eligibility(cow):
 
         if lactation.lactation_stage == LactationStageChoices.ENDED:
             raise ValidationError("Cannot add milk entry, Previous Lactation Ended!", code="previous_lactation_ended")
+        
+    @staticmethod
+    def validate_storage_and_usage(storage_location, usage):
```
Comment (peter-evance) [inline production/validators.py:186]: Remove this validator, or comment it out. For now its unnecessary.

**Review by peter-evance — CHANGES_REQUESTED** [review]
Check on these minor issues...

**Comment by mamesareagan** [comment]
Pushing in a few

**Comment by peter-evance** [comment]
> Pushing in a few

Ok.

**Pull request closed without merging**